api/v1alpha5/conversion.go

```diff
@@ -451,7 +453,7 @@ func Convert_v1alpha8_OpenStackMachineSpec_To_v1alpha5_OpenStackMachineSpec(in *
 		return err
 	}
 
-	if in.ServerGroup != nil {
+	if in.ServerGroup != nil && in.ServerGroup.ID != "" {
```

Comment:
Why do we care? If `in.ServerGroup.ID == ""`, then we can still set `out.ServerGroupID`, it'll have a default value anyway.

Reply:
`out.ServerGroupID` is already an empty string by default indeed. Yes we can remove that conditional.